### DISCUSSION TITLE
Forcer le needConsent = false pour un service existant avec Tac Pro en mode automatique

### DISCUSSION THREAD

**Discussion by tromeurf** [AmauriC/tarteaucitron.js#1193] [Q&A] [answered]
Bonjour,
j'ai eu beau chercher et retourner la question dans tous les sens, je n'arrive pas à Déclarer le needConsent à false pour le service hcaptcha avec tac pro en mode auto. Pour info j'utilise le plugin wordpress.
Quelqu'un a une solution ? une idée ?
Merci beaucoup pour votre aide

**Comment by tromeurf** [comment]
@AmauriC , une piste ?
Notre service juridique insiste pour que Hcaptcha soit déclaré comme service nécessaire au fonctionnement, 
donc affiché dans la modale des service comme service obligatoire, et pas bloqué par le mode auto.

**Reply by AmauriC** [reply]
Il est possible de contourner le blocage en ajoutant ce tag js:

```
window.addEventListener('tac.root_available', function() {
      tarteaucitron_block.unblock(/hcaptcha\.com\/1\/api\.js/);
});
```

Plus une règle CSS pour retirer la ligne du panneau:

```
li#hcaptchaLine {
    display: none;
}
```

**Reply by tromeurf** [reply]
Merci pour ta réponse rapide @AmauriC  !
En effet, mon captcha s'affiche bien ! Merci !

L'option needConsent fait quoi exactement ? (l'ai l'impression que ça active par défaut le consentement mais qu'on peut le refuser après coup).
Ce que j'aimerai, c'est avoir mon service hcaptcha visible mais sans pouvoir toggle le bouton de consentement.
Est-ce possible ?

Merci beaucoup 🙏

**Reply by AmauriC** [reply, marked as answer]
C'est bien ça pour le `needConsent`, le service est par défaut accepté. C'est principalement pour des outils de mesure d'audience comme Matomo. hcaptcha doit théoriquement recevoir le consentement de l'utilisateur pour être chargé.

Ces règles CSS masqueront le bouton refuser et accepter sera toujours comme accepté:
```
button#hcaptchaAllowed {
    opacity: 1!important;
}

button#hcaptchaDenied {
    opacity: 0!important;
}
```

**Reply by tromeurf** [reply]
Merci beaucoup pour ta réactivité et tes explications ! 

Pour le bouton denied, il faut carrément mettre un display none
`
button#hcaptchaDenied {
    display: none!important;
}
`

**Reply by aquerol2** [reply]
Bonjour,
est-ce que c'est bien pris en compte si l'utilisateur sélectionne "interdire tous les cookies" ? Il me semble que même si le `needConsent` est à `false` et que le bouton "interdire" pour ce service est caché l'utilisateur peut quand même modifier sa valeur

**Reply by AmauriC** [reply]
Le `needConsent` c'est pour le chargement par défaut mais ça n'empêche pas l'utilisateur de le refuser. 

Pour exclure un service du mode auto il faut faire différemment. 

Avec le nom du service je peux donner la marche à suivre mais ⚠️ attention, ce sera une non conformité et je ne le recommande pas.

**Reply by aquerol2** [reply]
Je veux bien la marche à suivre stp.
Il s'agit de piwik pro. On a anonymisé les données et on peut donc autoriser le service par défaut et être conforme.

**Reply by AmauriC** [reply]
Ok 👍

Ajoutez avant le tag tarteaucitron:

```
<script type="text/javascript">
window.tarteaucitron_whitelist = [];
window.tarteaucitron_whitelist.push(/\.piwik\.pro/);
</script>
```

**Reply by aquerol2** [reply]
Merci ! 👍
